src/odemis/driver/pwrmccdaq.py

```
@@ -364,6 +379,266 @@ def _update_power(self, value: List[float]):
+            unit = axis_cfg.get("unit", "")
+            duration = axis_cfg["duration"]
+
+            if low == high:
+                raise ValueError(f"Axis '{axis_name}' 'low' and 'high' positions must be different")
+            if not 0 <= duration < 1000:
+                raise ValueError(f"Axis '{axis_name}' duration {duration} should be between 0 and 1000 s")
+            try:
+                do_port, _ = parent.channel_to_port(do_ch)
+                do_ports.add(do_port)
+            except ValueError:
+                raise ValueError(f"Axis '{axis_name}' has invalid 'do' channel {do_ch}")
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)